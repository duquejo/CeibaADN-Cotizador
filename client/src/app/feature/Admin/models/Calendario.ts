export interface Calendario {
  id?: number;
  nombre: string;
  descripcion: string;
  festivos: Array<string>;
}
