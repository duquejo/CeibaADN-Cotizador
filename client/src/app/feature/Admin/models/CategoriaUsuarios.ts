export interface CategoriaUsuarios {
  id?: number;
  nombre: string;
  descripcion: string;
  valorAlta: number;
  valorBaja: number;
}
