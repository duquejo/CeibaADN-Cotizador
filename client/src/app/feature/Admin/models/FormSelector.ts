export interface ISelectionOptions {
    label: string;
    value: string;
}
