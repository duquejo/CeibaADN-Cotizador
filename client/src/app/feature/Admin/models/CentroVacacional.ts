import { Calendario } from "./Calendario";
import { CategoriaUsuarios } from "./CategoriaUsuarios";

export interface CentroVacacional {
  id?: number;
  nombre: string;
  descripcion: string;
  calendarios: Array<Calendario>;
  categoriaUsuarios: Array<CategoriaUsuarios>;
  calendarioActivo: number;
}
