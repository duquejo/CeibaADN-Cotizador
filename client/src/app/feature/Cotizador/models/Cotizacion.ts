import { CategoriaUsuarios } from 'app/feature/Admin/models/CategoriaUsuarios';
import { CentroVacacional } from '../../Admin/models/CentroVacacional';

export interface Cotizacion {
  id?: number;
  codigo?: string;
  centroVacacional: number | CentroVacacional;
  categoriaUsuarios: number | CategoriaUsuarios;
  fechaInicio: string;
  personas: number;
  fechaFin: string;
  total?: number;
}
