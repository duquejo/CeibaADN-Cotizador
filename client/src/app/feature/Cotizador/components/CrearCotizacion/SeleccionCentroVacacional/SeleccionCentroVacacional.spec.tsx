import * as React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import { SeleccionCentroVacacional } from './index';
import { SinonStub, stub } from 'sinon';
import { CentroVacacional } from '../../../../Admin/models/CentroVacacional';

describe('SeleccionCentroVacacional Test', () => {

  let componentWrapper: ShallowWrapper;

//   const handlerClick = jest.fn();

  let componentProps: React.ComponentProps<typeof SeleccionCentroVacacional> & {
    elementos: CentroVacacional[];
    handlerSeleccion: SinonStub;
  };

  beforeEach( () => {
        componentProps = {
        handlerSeleccion: stub(),
        elementos: [{
                id: 1,
                nombre: 'Centro de prueba',
                descripcion: 'Descripción de prueba',
                calendarios: [],
                categoriaUsuarios: [],
                calendarioActivo: 1
            }]
        };
        componentWrapper = shallow(<SeleccionCentroVacacional { ...componentProps } />);
        jest.clearAllMocks();
    });  

  it('Debería renderizar el componente', () => {
    expect(componentWrapper).toMatchSnapshot();
  });

//   it('Debería activar el elemento al hacerle click', () => {

//     const slideBefore = componentWrapper.find('SwiperSlide').first();

//     console.log( 'before', slideBefore.first().props() );

//     slideBefore.simulate( 'click', { 
//         currentTarget: () => ' selected', 
//         centroVacacional: componentProps.elementos[0]
//      } );
    
//     const slideAfter = componentWrapper.find('SwiperSlide').first();

//     console.log( 'after', slideAfter.first().props() );

//     // expect( slideBefore.prop('className') ).toContain( ' selected' );
//   });

});
