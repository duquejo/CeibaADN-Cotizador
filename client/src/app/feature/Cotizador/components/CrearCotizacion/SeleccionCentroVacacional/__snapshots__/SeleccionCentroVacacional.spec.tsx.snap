// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeleccionCentroVacacional Test Debería renderizar el componente 1`] = `
<div
  className="swiperContainer"
>
  <Swiper
    pagination={true}
    slidesPerView={3}
    spaceBetween={20}
  >
    <SwiperSlide
      key="1"
      onClick={[Function]}
    >
      <article
        className="slideContent"
      >
        <div
          className="slideTitle"
        >
          Centro de prueba
        </div>
        <div
          className="slideDescripcion"
        >
          Descripción de prueba
        </div>
      </article>
    </SwiperSlide>
  </Swiper>
</div>
`;
